### ISSUE TITLE
Synergy mouse freezes briefly and resets when used with WebEx.

### ISSUE THREAD

**Issue by elcman** [symless/synergy#5939]
### Operating Systems ###

Server: Windows 10 Home v. 1607 (OS Build 14393.953)
Client: Windows 10 Professional v. 1607 (OS Build 14393.953)

### Synergy Version ###

1.8.8-stable-c30301e

### Steps to reproduce bug ###

1. Launch WebEx on client system
2. During an active WebEx session, move mouse onto client screen
3. Intermittent stuttering/hard freeze on the whole system
4. Mouse resets to the middle of Server screen, client picks up again
5. Repeat from 2, until WebEx is closed
6. Mouse works as normal once

### Other info ###

* I have had some updates where this didn't seem to be a problem, but then it starts up again. I thought the latest version had addressed it, but it worked only for a couple of minutes before it started hanging again.
* WebEx is grabbing the mouse activity that seems to be incompatible with Synergy.
* This is regardless of if I'm presenting, viewing a presentation, and/or remotely controlling another computer.
* Typing doesn't seem to be a problem. I can move my mouse over quickly before the buffer overloads or freezes and I can type without issue while the mouse is on the screen.
* This issue seemed to appear around ver. 1.7.x. Before then it didn't seem to be an issue. I have been tempted to roll back and see if that makes a difference.

I live in WebEx, working from home. This is highly impacts me... but I haven't seen much from others around this. So, I kinda expect this is a low priority. Hopefully, another fix helps this secondarily.

**Comment by XinyuHou** [comment]
@elcman 
This sounds like a network issue. What is your bandwidth? Could you also send us some logging with debug level set to Debug.

**Comment by xpac1985** [comment]
I experienced the same problem yesterday. I was wondering if it is because of WebEx or Citrix Receiver, or if both are involved. I had WebEx running on the server and Citrix on the client - did you do the same @elcman? Will append my logs here later.

**Comment by xpac1985** [comment]
ATM, I can only offer non-debug logs, as I cannot reproduce at the moment.

What happens is - cursor freezes, no movement on client or server possible for like 5 seconds. Afterwards, cursor resets to center server display. Reconnection starts, after a few seconds, Synergy can be used again. It's definetely not a network infrastructure issue - both devices on LAN, and this only happens with the WebEx(/maybe Citrix) constellation for me, besides everything is fine.

It always looks like this:

`...many leaving screen - entering screen events here...
[2017-03-30T16:27:40] INFO: leaving screen
[2017-03-30T16:27:40] INFO: entering screen
[2017-03-30T16:29:09] NOTE: server is dead
[2017-03-30T16:29:09] INFO: leaving screen
[2017-03-30T16:29:10] WARNING: failed to connect to server: server is not responding
[2017-03-30T16:29:11] NOTE: connecting to '$servername': $serverIP:24800
[2017-03-30T16:29:11] INFO: OpenSSL 1.0.2 22 Jan 2015
[2017-03-30T16:29:11] NOTE: server fingerprint: $fingerprint
[2017-03-30T16:29:11] INFO: connected to secure socket
[2017-03-30T16:29:11] INFO: server ssl certificate info: /CN=Synergy
[2017-03-30T16:29:11] INFO: AES256-GCM-SHA384       TLSv1.2 Kx=RSA      Au=RSA  Enc=AESGCM(256) Mac=AEAD
[2017-03-30T16:29:11] NOTE: connected to server`

From then, everything works fine again...

**Comment by kutschinsl** [comment]
I have the same issue, even the local hardware mouse and keyboard on the client freezes for that period.  I would agree it doesn't seem to be a network issue, and only happens with WebEx.  It also happens for me with WebEx running even with no other participants joined.  Would be great to have a fix for this as I have to shut down synergy before every meeting at the moment!

I've attached client and server debug captures.  WebEx was running on the client.
[server-debug.zip](https://github.com/symless/synergy/files/891245/server-debug.zip)
[client-debug.zip](https://github.com/symless/synergy/files/891328/client-debug.zip)

**Comment by XinyuHou** [comment]
Guys, this looks like a duplicate of #5857. I will close this and merge it into #5857

Based oin your log, it's connection issue when using SSL. And there is not any big data in clipboard.

**Comment by xpac1985** [comment]
I'm pretty sure this is not a duplicate. It happens ONLY with Webex and happens no matter if SSL is disabled or enabled.

The Windows event log shows "The Synergy service terminated unexpectedly. It has done this 1 time(s). The following corrective action will be taken in 1000 milliseconds: Restart the service."

Server log:
`[2017-04-28T14:50:26] INFO: switch from "$CLIENT" to "$SERVER" at 1918,382
[2017-04-28T14:50:26] INFO: entering screen
[2017-04-28T14:53:54] INFO: switch from "$SERVER" to "$CLIENT" at 0,244
[2017-04-28T14:53:54] INFO: leaving screen
[2017-04-28T14:53:54] DEBUG: dropped bogus delta motion: +977,-313
[2017-04-28T14:54:03] NOTE: client "$CLIENT" has disconnected
[2017-04-28T14:54:03] INFO: jump from "$CLIENT" to "$SERVER" at 960,540
[2017-04-28T14:54:03] INFO: entering screen
[2017-04-28T14:54:03] DEBUG: active sides: 4
[2017-04-28T14:54:04] NOTE: accepted client connection
[2017-04-28T14:54:04] DEBUG: received client "$CLIENT" info shape=0,0 3968x1152 at 0,0
[2017-04-28T14:54:04] DEBUG: active sides: 4
[2017-04-28T14:54:04] NOTE: client "$CLIENT" has connected
[2017-04-28T14:54:05] DEBUG: accepted ipc client connection
[2017-04-28T14:54:06] DEBUG: start ipc handle data
[2017-04-28T14:54:06] DEBUG: ipc read: IHEL
[2017-04-28T14:54:06] DEBUG: finished ipc handle data
[2017-04-28T14:54:06] DEBUG: ipc hello, type=gui
[2017-04-28T14:54:06] INFO: watchdog status: ok
[2017-04-28T14:54:11] INFO: switch from "$SERVER" to "$CLIENT" at 0,888
`

Client log:
`[2017-04-28T14:50:26] INFO: entering screen
[2017-04-28T14:50:26] INFO: leaving screen
[2017-04-28T14:53:54] INFO: entering screen
[2017-04-28T14:54:03] NOTE: server is dead
[2017-04-28T14:54:03] INFO: leaving screen
[2017-04-28T14:54:03] WARNING: failed to connect to server: server is not responding
[2017-04-28T14:54:03] DEBUG: retry in 1 seconds
[2017-04-28T14:54:04] NOTE: connecting to '$SERVER': 192.168.23.31:24800
[2017-04-28T14:54:04] DEBUG: desk Default window is 0x0047025a
[2017-04-28T14:54:04] DEBUG: switched to desk "Default"
[2017-04-28T14:54:04] DEBUG: desktop is now accessible
[2017-04-28T14:54:04] DEBUG: open clipboard
[2017-04-28T14:54:04] DEBUG: empty clipboard
[2017-04-28T14:54:04] DEBUG: close clipboard
[2017-04-28T14:54:04] DEBUG: open clipboard
[2017-04-28T14:54:04] DEBUG: empty clipboard
[2017-04-28T14:54:04] DEBUG: close clipboard
[2017-04-28T14:54:04] NOTE: connected to server
[2017-04-28T14:54:11] INFO: entering screen
`

This is REALLY unnerving. Especially the fact that the server completely hangs for several seconds (audio/video playback completely hangs), the CPU load rises (I can hear my fans spin up).

**Comment by kutschinsl** [comment]
I agree this is not an SSL issue.  I have experience the same issue with SSL and Clipboard sharing OFF.

**Comment by NomNomGood** [comment]
Was having the same exact issues via WEBEX and Synergy 1.8.8-stable-c30301e.  Whenever I launch WEBEX for presentation the mouse would freeze up every few seconds for a few seconds then resume.  On top of that, I have to go through VPN for one system and that sometimes disconnects my local shares.  So when WEBEX and Synergy acts up, it just stops allowing me to connect from the main system to the VPN'd laptop.  I had to plug in a separate mouse and keyboard or use my laptop keyboard whenever WEBEX was launched.  I searched a while back but did not find anything so I searched again today.  Disabling SSL worked for me.  I tested it just earlier through two lengthy WEBEX presentations without SSL enabled on all my systems and had no further freezing issues.  I know other folks here have said disabling SSL did not resolve their issue, but just wanted to say thanks at least for me in that it seems to be a resolution.

**Comment by NomNomGood** [comment]
Spoke too soon, I guess rebooting the laptop caused the issue to return.  Oh well, have to keep digging or find a better way.

**Comment by cigoldstein** [comment]
I'm having the exact same problem.  This occurs only when WebEx is running, and immediately stops when I close the WebEx application.  This issue occurs whether WebEx is running on the server or the client; the machine running the WebEx will report as "dead" every few seconds.

**Comment by keatre** [comment]
Having the same issue - have submitted a request to Synergy support, but they only offered me a refund... I love the product and would like to keep using, but need it to work for WebEx.  I currently have a second mouse plugged into my client machine for when I have to start a WebEx.

1.8.8-stable

**Comment by xpac1985** [comment]
I've been spammed with like 4 mails regarding paying money to join the NEW SYNERGY BETA omg!!!1

And here we are, still stuck with a game-breaking problem for more than half a year, almost without any dev responses. Makes me really regret paying money für Synergy, in the meantime I uninstalled it in favor of the free Mouse without Borders from Microsoft, which works with a lot less issues.

**Comment by mjo44** [comment]
Had the same exact issue with1.8.8-stable-c30301e (no SSL) on Windows 10 Pro 64bit running Webex, intermittent behavior, mouse would freeze 3-5 seconds at a time. Would depend on overall network traffic though as far as frequency of lag. But the fix for me was upgrading to the latest code on Synergy website, 1.8.8-stable-25a8cb2. Haven't rebooted yet to see check consistency, but confidence is high.

**Comment by keatre** [comment]
I just tested this with Synergy 2.0.1 and can confirm the issue still persists.  However, due to how Synergy 2 handles server / client, it actually makes this issue quite a bit worse. As a workaround for 1.8.8, I had a mouse plugged into the client machine that runs the WebEx's, and would use that when having to navigate the client while a WebEx was running.  However, now with 2.0.1, Synergy confuses itself - thinking I want to use that mouse as the server, switches it, freezes again, and basically locks me out of all movements to all connected computer (meaning I cannot navigate to my Macbook as client #2). 

Have reported back to Synergy support, but not sure they're handling this as a priority.  I have since downgraded back to 1.8.8 for the time being.
